src/main/java/spoon/metamodel/MetamodelProperty.java

```diff
@@ -600,7 +601,7 @@ private	static ContainerKind containerKindOf(Class<?> valueClass) {
 		if (Map.class.isAssignableFrom(valueClass)) {
 			return ContainerKind.MAP;
 		}
-		if (Set.class.isAssignableFrom(valueClass)) {
+		if (Set.class.isAssignableFrom(valueClass) || Collection.class.isAssignableFrom(valueClass)) {
```

Comment:
Set is a subtype of Collection so only the second part is necessary
```suggestion
		if (Collection.class.isAssignableFrom(valueClass)) {
```